src/lib/services/proxy-service.ts

```diff
@@ -85,15 +85,15 @@ export class ProxyService {
     }
 
     async registerProxyMetrics(
-        token: ApiUser,
+        token: IApiUser,
         metrics: ClientMetricsSchema,
         ip: string,
     ): Promise<void> {
         ProxyService.assertExpectedTokenType(token);
 
         const environment =
             this.services.clientMetricsServiceV2.resolveMetricsEnvironment(
-                token,
+                token as ApiUser,
```

Comment:
Not so easy to change cause we're testing `user instanceof ApiUser` inside resolveMetricsEnvironment. We can't just use isAPI because we also have users who are APIs (service accounts IIRC)